x/concentrated-liquidity/math/tick.go

```diff
@@ -29,22 +29,36 @@ func TicksToSqrtPrice(lowerTick, upperTick int64) (osmomath.BigDec, osmomath.Big
 // TickToSqrtPrice returns the sqrtPrice given a tickIndex
 // If tickIndex is zero, the function returns osmomath.OneDec().
 // It is the combination of calling TickToPrice followed by Sqrt.
-func TickToSqrtPrice(tickIndex int64) (osmomath.BigDec, osmomath.BigDec, error) {
+func TickToSqrtPrice(tickIndex int64) (osmomath.BigDec, error) {
 	priceBigDec, err := TickToPrice(tickIndex)
 	if err != nil {
-		return osmomath.BigDec{}, osmomath.BigDec{}, err
+		return osmomath.BigDec{}, err
 	}
 
-	// It is acceptable to truncate here as TickToPrice() function converts
-	// from osmomath.Dec to osmomath.BigDec before returning.
-	price := priceBigDec.Dec()
+	// N.B. at launch, we only supported price range
+	// of [tick(10^-12), tick(MaxSpotPrice)].
+	// To maintain backwards state-compatibility, we use the original
+	// math based on 18 precision decimal on the at the launch tick range.
+	if tickIndex >= types.MinInitializedTick {
+		// It is acceptable to truncate here as TickToPrice() function converts
+		// from osmomath.Dec to osmomath.BigDec before returning specifically for this range.
+		// As a result, there is no data loss.
+		price := priceBigDec.Dec()
```

Comment:
To clarify the soundness of this truncation: have we confirmed that mononicity holds across this boundary?

i.e. even if the bigdec progression is monotonic, if the truncated value is >0 (as would be the case in all scenarios where MinInitializedTick does not correspond exactly to a value that ends with 18 zeroes in BigDec form), then the overall price progression would be non-monotonic since the truncation would lead to a dip in price for a higher tick.

If this is a non-issue, we should explain why in comments. Otherwise, it's a violation of mononicity and we should discuss best way ti address it.

Reply:
Good point.

This is implicitly covered by:
https://github.com/osmosis-labs/osmosis/blob/3e516838b2db4a1f4b95ae5c64b058350b6d68fd/x/concentrated-liquidity/math/tick_test.go#L128-L139

and

https://github.com/osmosis-labs/osmosis/blob/3e516838b2db4a1f4b95ae5c64b058350b6d68fd/x/concentrated-liquidity/math/tick_test.go#L852

The reason why the second test applies is because `CalculateSqrtPriceToTick` computes tick-to-sqrt below, at and above the given tick:
https://github.com/osmosis-labs/osmosis/blob/3e516838b2db4a1f4b95ae5c64b058350b6d68fd/x/concentrated-liquidity/math/tick.go#L287-L290

Please let me know if this addresses your concern